src/dreamer/surprisal.py

```
@@ -0,0 +1,470 @@
+                "📊 Filtered statistics: "
+                + f"min={filtered[-1].surprisal:.3f}, "
+                + f"max={filtered[0].surprisal:.3f}, "
+                + f"mean={sum(s.surprisal for s in filtered) / len(filtered):.3f}"
+            )
+        else:
+            logger.info("No observations exceeded the surprisal threshold")
+
+        return filtered
+
+    except Exception as e:
+        logger.error(f"Surprisal sampling failed: {e}", exc_info=True)
```

Ruff (0.14.10)

Logging `.exception(...)` should be used instead of `.error(..., exc_info=True)`

(G201)